src/pages/TestDrive/TestDrive.tsx

```diff
@@ -0,0 +1,117 @@
+import { Button, Container, Form, Card } from "react-bootstrap";
+import { useLocation, useParams, Link, useNavigate } from "react-router-dom";
+import React, { useState } from "react";
+import "./TestDrive.scss";
+function TestDrive() {
+  let navigate = useNavigate();
+  const [mobileValid, setMobileValid] = useState(false);
+  const [emailValid,setEmailValid] = useState(false);
+  var pattern = /^(\+\d{1,3}[- ]?)?\d{10}$/;
+  let emailPattern = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/
+  const [validated, setValidated] = useState(false);
+
+  const handleSubmit = (event: any) => {
+    const form = event.currentTarget;
+    event.preventDefault();
+    event.stopPropagation();
+    if (form.checkValidity() === true) {
+      console.log("True")
+    }
+    setValidated(true);
+  };
+
+  const handleChange = (event: any) => {
+    const {name,value} = event.target
+    if(name==="contact"){
+        if (!(!value || pattern.test(value) === false)) {
```

Comment:
if you want to check if there is some value in it. simply check that at the top, i.e. after line 24, 
```
if(!value){
  return;
}
```

and in here simply do `if(pattern.test(value))` instead of complicating things here